### PR TITLE
delete work orders on MO cancel

### PR DESCRIPTION
**Pull request by sfenig**
Note: I haven't been able to verify that MO's created by Replenish To Order (name in community edition I have)/Make To Order would successfully delete when the origin is cancelled.  I can't seem to get the child MOs to cancel at all.  Manually cancelling them does also delete the work orders FWIW.

### CODE REVIEW

**Comment by sfenig** [comment]
> Note: I haven't been able to verify that MO's created by Replenish To Order (name in community edition I have)/Make To Order would successfully delete when the origin is cancelled. I can't seem to get the child MOs to cancel at all. Manually cancelling them does also delete the work orders FWIW.
There is not propagating cancel.  Confirmed delete also runs cancel, so bulk delete from the list page will also delete pico work orders.

**Comment by zacnelson** [comment]
While testing today, I tried to click on "cancel" on an existing WO I had created prior to updating the pico module to this branch. It went ahead and cancelled but at the time it didn't appear to push anything to the api on my server. I think I had already deleted WO locally, but was expecting some sort of API logs.

I then created a fresh WO and was able to cancel it no problem.

**Comment by zacnelson** [comment]
I tested out the cascaded cancel with a Made to Order work order. When I clicked on cancel I didn't get any sort of errors, but there was a flag next to the 1-level-down assembly's work order that said this:
image

File: pico_mrp/models/mrp.py [hunk 1]
```diff
@@ -28,6 +28,17 @@ def action_confirm(self):
             production._pico_create_work_orders()
         return res
 
+    def _pico_delete_work_orders(self):
+        for work_order in self.pico_work_order_ids:
+          work_order.pico_delete()
```
Comment (zacnelson) [inline pico_mrp/models/mrp.py]: Something odd with indentation here.

File: pico_mrp/models/mrp.py [hunk 2]
```diff
@@ -162,6 +173,13 @@ def _pico_create(self):
             'state': 'running',
         })
 
+    def pico_delete(self):
+        self._pico_delete()
+
+    def _pico_delete(self):
+        api = pico_api(self.env)
+        api.delete_work_order(self.pico_id)
```
Comment (zacnelson) [inline pico_mrp/models/mrp.py:181]: can these just go up in the above public function?

**Comment by sfenig** [comment]
> I tested out the cascaded cancel with a Made to Order work order. When I clicked on cancel I didn't get any sort of errors, but there was a flag next to the 1-level-down assembly's work order that said this:
> image

Yeah this is what I saw as well, I think it's the expected behavior.  I did confirm that you can bulk delete from the list page and it would also cancel.

**Review by zacnelson — COMMENTED** [review]
Ugh, github comments are confusing. Comments published now.

File: pico_mrp/models/mrp.py (hunk 2 above)
Comment (sfenig) [reply pico_mrp/models/mrp.py:181]: I figured this was some odoo convention.  I was following @jaredkipe's code on pico_create above.

File: pico_mrp/models/mrp.py (hunk 1 above)
Comment (sfenig) [reply pico_mrp/models/mrp.py]: fixed

File: pico_mrp/models/mrp.py (hunk 2 above)
Comment (zacnelson) [reply pico_mrp/models/mrp.py:181]: OK, can leave it for now. Would be curious to learn from @jaredkipe if there's a convention to follow and when to do so.

**Pull request merged**